File: Backend/src/models/address_tel_numbers.py
# Backend/src/models/address_tel_numbers.py
# Definition of the AddressTelNumbers model that connects addresses and telefone numbers.

# Author: Valentin Haas, 2024

# System imports
from typing import Optional

# Library imports
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_pascal
from sqlalchemy import Column, Integer, Boolean
from sqlalchemy.ext.declarative import declarative_base

# Constants
BASE = declarative_base()


class AddressTelNumbersTable(BASE):
    """Definition of the AddressTelNumbers model for the database."""

    __tablename__ = "AddressTelNumbers"

    Address = Column(Integer, primary_key=True, foreign_key="Addresses.Id")
    """The unique identifier of the address."""

    TelNumber = Column(Integer, primary_key=True, foreign_key="TelefoneNumbers.Id")
    """The unique identifier of the telefone number."""

    IsPrimary = Column(Boolean, nullable=False, default=True)
    """Indicates whether the telefone number is the primary telefone number for the address."""


class AddressTelNumber(BaseModel):
    """Definition of the AddressTelNumber model for the API."""

    model_config = ConfigDict(populate_by_name=True, alias_generator=to_pascal)

    address: int
    """The unique identifier of the address."""

    tel_number: int
    """The unique identifier of the telefone number."""

    is_primary: Optional[bool] = True
    """Indicates whether the telefone number is the primary telefone number for the address."""

File: Backend/src/models/addresstelnumbers.go
// Backend/src/models/addresstelnumbers.go
// Definition of the AddressTelNumbers model that connects addresses and telephone numbers.

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// AddressTelNumbers represents the connection between addresses and telephone numbers.
type AddressTelNumber struct {
	gorm.Model  // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields
	Address     Address
	AddressID   uint            `gorm:"primaryKey"` // The unique identifier of the address, acts as part of composite primary key.
	TelNumber   TelephoneNumber // Referenced TelephoneNumer Object
	TelNumberID uint            `gorm:"primaryKey"` // The unique identifier of the telephone number, acts as part of composite primary key.
	IsPrimary   bool            `gorm:"not null"`   // Indicates whether it's the primary telephone number for the address.
}
